src/components/tools/NewToolRequestDialog.js

```diff
@@ -0,0 +1,180 @@
+        }
+    );
+
+    const handleSubmit = (values, { props }) => {
+        //test_data_path and cmd_line are required by service. Most of the times those info. can be
+        // found in the source repo. So copy the source url to those fields if not provided by user.
+        if (!values.test_data_path) {
+            values.test_data_path = values.source_url;
+        }
+        if (!values.cmd_line) {
+            values.cmd_line = values.source_url;
+        }
+        const submission = { ...values };
+        if (submitRequestStatus !== constants.LOADING) {
+            submitNewToolRequest({ submission });
+        }
+    };
+
+    const validateUrl = (value) => {
+        let error;
+        if (!constants.URL_REGEX.test(value)) {
+            error = t("validationErrMsgURL");
+        }
+        return error;
+    };
+
+    return (
+        <DEDialog
+            open={open}
+            onClose={onClose}
+            baseId={build(baseId, ids.TOOL_REQUEST.DIALOG)}
+            title={t("newToolRequestDialogHeading")}
+        >
+            {requestError && (
+                <ErrorTypographyWithDialog
+                    errorObject={requestError}
+                    errorMessage={t("toolRequestError")}
+                    baseId={baseId}
+                />
+            )}
+            <Formik
+                initialValues={{
+                    name: "",
+                    description: "",
+                    source_url: "",
+                    version: "",
+                    documentation_url: "",
+                    cmd_line: "",
+                    test_data_path: "",
+                }}
+                enableReinitialize={true}
+                onSubmit={handleSubmit}
+            >
+                {() => {
+                    return (
+                        <Form>
+                            <Field
+                                name="name"
+                                label={t("toolNameLabel")}
+                                required={true}
+                                margin="dense"
+                                id={build(baseId, ids.TOOL_REQUEST.NAME)}
+                                component={FormTextField}
+                            />
+                            <Field
+                                name="description"
+                                label={t("toolDescLabel")}
+                                required={true}
+                                margin="dense"
+                                id={build(baseId, ids.TOOL_REQUEST.DESCRIPTION)}
+                                component={FormMultilineTextField}
+                            />
+                            <Field
+                                name="source_url"
+                                label={t("toolSrcLinkLabel")}
+                                required={true}
+                                margin="dense"
+                                validate={validateUrl}
```

Comment:
This is the only validation check I see in this form.

I haven't tried this locally yet, but does this mean the user can submit this form without filling in the other required fields? If that's the case, then the error message that tells the user to "Please try again later" is misleading if the service is returning an error about a missing form field.

In our other forms we have validation checks that will return a `Required` error message if a required field is empty, which also prevents the form submitting to the service.

Reply:
Hmm..There are required fields in this form. Only the `source_url` field needs additional validation for a valid url. See screenshots below.

![image](https://user-images.githubusercontent.com/1250790/104743769-0d5e9280-571a-11eb-824e-949e4fc47cd2.png)

![image](https://user-images.githubusercontent.com/1250790/104743817-1e0f0880-571a-11eb-87ec-4846a4ac5fa8.png)

Reply:
I see the required fields are marked as required by MUI, but I don't think Formik knows about that, and the question is what happens when you try to submit the form without filling in those fields? I think that's what you are showing in the first screenshot? Then what happens if you "touch" a required field without entering anything, then focus on another field?

Without a validation function to return a `Required` error message, I don't think the required fields will be highlighted in red, like the URL field in the second screenshot. Note that's how it works in the `Add Tool` form, if you focus on the tool name, then focus on another field:
![Add Tool form errors](https://user-images.githubusercontent.com/996408/104762804-3125d680-5722-11eb-8113-a4324383b2cc.png)

The App Launch form also behaves this way, as well as highlighting required fields when submitting the form even without touching any fields:
![App Launch form error](https://user-images.githubusercontent.com/996408/104763127-ae514b80-5722-11eb-8188-f0d5eefceda6.png)


We should keep our form validation behavior consistent.

Reply:
Yup. The first screenshot shows what happens when you try to submit the form without entering anything. And right now nothing happens when focus out of a field. I just copied these over from DE so I checked there now and it has the same behavior. But I agree that it needs to consistent. I will update the logic.

Reply:
Ok this should be done!.